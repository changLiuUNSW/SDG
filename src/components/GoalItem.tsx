import { Ionicons } from '@expo/vector-icons';
import React from 'react';
import styled from 'styled-components/native';

import { Goal } from '../models/Goal';

const Container = styled.TouchableOpacity`
  margin-horizontal: 16;
  justify-content: center;
  align-items: center;
  border-color: #e6e9ee;
  border-radius: 8;
  border-width: 1;
  background-color: #ffff;
  margin-vertical: 8;
`;

const Text = styled.Text`
  text-align: center;
  font-size: 18;
`;

interface Props {
  goal: Goal;
  onPress?: () => void;
}
const GoalItem = ({ goal, onPress }: Props) => (
  <Container testID={`goalItem-${goal.code}`} onPress={onPress} disabled={!onPress}>
    <Text>{`${goal.code}. ${goal.title}`}</Text>
    <Ionicons name="ios-people" size={80} />
  </Container>
);

export default GoalItem;
