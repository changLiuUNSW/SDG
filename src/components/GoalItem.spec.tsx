import * as React from 'react';
import renderer from 'react-test-renderer';

import GoalItem from './GoalItem';
import { shallow } from 'enzyme';

const setup = (propertiesOverride?: any) => {
  const props = Object.assign(
    {
      goal: {
        code: '13',
        title: 'test 13'
      }
    },
    propertiesOverride
  );

  const wrapper = () => <GoalItem {...props} />;

  return {
    props,
    wrapper
  };
};

it('should render the component correctly', () => {
  const { wrapper } = setup();
  const component = renderer.create(wrapper());
  expect(component.toJSON()).toMatchSnapshot();
});

it('should disabled if noPress pass via prop', () => {
  const { wrapper } = setup();
  const component = shallow(wrapper());
  expect(component.prop('disabled')).toBeTruthy();
});

it('onPress should be called when passed via prop', () => {
  const onPress = jest.fn();
  const { wrapper } = setup({
    onPress
  });
  const component = shallow(wrapper());
  component.props().onPress();
  expect(onPress).toBeCalled();
});
