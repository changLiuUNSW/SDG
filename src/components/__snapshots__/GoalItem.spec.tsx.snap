// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the component correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffff",
      "borderColor": "#e6e9ee",
      "borderRadius": 8,
      "borderWidth": 1,
      "justifyContent": "center",
      "marginHorizontal": 16,
      "marginVertical": 8,
      "opacity": 1,
    }
  }
  testID="goalItem-13"
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 18,
          "textAlign": "center",
        },
      ]
    }
  >
    13. test 13
  </Text>
  <Text />
</View>
`;
